File: TCP/Server/daytime_server.py
import socket
from datetime import datetime

# Server Configuration
host = '127.0.0.1'  # Localhost
port = 12345        # Arbitrary non-privileged port

# Create a TCP/IP socket
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as server_socket:
    server_socket.bind((host, port))
    server_socket.listen(1)  # Listen for incoming connections

    print(f"Server started at {host} on port {port}")
    print("Waiting for a connection...")

    while True:
        # Wait for a connection
        connection, client_address = server_socket.accept()

        with connection:
            print(f"Connected by {client_address}")

            # Get the current date and time
            current_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")

            # Send the current date and time to the client
            connection.sendall(current_time.encode())

            print(f"Sent to {client_address}: {current_time}")

File: TCP/Server/daytime_server.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <unistd.h>
#include <arpa/inet.h>
#include <time.h>

#define PORT 12345
#define BUFFER_SIZE 1024

int main() 
{
    int server_fd, new_socket;
    struct sockaddr_in address;
    int addrlen = sizeof(address);
    char buffer[BUFFER_SIZE] = {0};
    time_t current_time;
    char *time_str;
    if ((server_fd = socket(AF_INET, SOCK_STREAM, 0)) == 0) 
    {
        perror("Socket failed");
        exit(EXIT_FAILURE);
    }
    address.sin_family = AF_INET;
    address.sin_addr.s_addr = INADDR_ANY;
    address.sin_port = htons(PORT);
    if (bind(server_fd, (struct sockaddr *)&address, sizeof(address)) < 0) 
    {
        perror("Bind failed");
        close(server_fd);
        exit(EXIT_FAILURE);
    }
    if (listen(server_fd, 3) < 0) 
    {
        perror("Listen failed");
        close(server_fd);
        exit(EXIT_FAILURE);
    }

    printf("Server started. Waiting for connections...\n");

    while (1) 
    {
        if ((new_socket = accept(server_fd, (struct sockaddr *)&address, (socklen_t*)&addrlen)) < 0) {
            perror("Accept failed");
            close(server_fd);
            exit(EXIT_FAILURE);
        }
        current_time = time(NULL);
        time_str = ctime(&current_time);
        send(new_socket, time_str, strlen(time_str), 0);
        printf("Sent to client: %s\n", time_str);

        close(new_socket);
    }

    close(server_fd);
    return 0;
}
